idaplugin/rematch/actions/match.py

```diff
@@ -71,60 +100,46 @@ def response_handler(self, file_version):
   def start_upload(self):
     self.functions = set(idautils.Functions())
 
-    self.pbar = QtWidgets.QProgressDialog()
     self.pbar.setLabelText("Processing IDB... You may continue working,\nbut "
                            "please avoid making any ground-breaking changes.")
     self.pbar.setRange(0, len(self.functions))
     self.pbar.setValue(0)
-    self.pbar.canceled.connect(self.cancel_upload)
-    self.pbar.rejected.connect(self.reject_upload)
     self.pbar.accepted.connect(self.accept_upload)
+    self.pbar.show()
 
-    self.timer = QtCore.QTimer()
     self.timer.timeout.connect(self.perform_upload)
     self.timer.start(0)
 
     return True
 
   def perform_upload(self):
-    try:
-      offset = self.functions.pop()
-    except KeyError:
-      self.timer.stop()
+    if not self.functions:
       return
 
-    try:
-      func = instances.FunctionInstance(self.file_version_id, offset)
-      self.instance_set.append(func.serialize())
-
-      if len(self.instance_set) >= 100:
-        network.delayed_query("POST", "collab/instances/",
-                              params=self.instance_set, json=True,
-                              callback=self.progress_advance)
-        self.instance_set = []
-        self.pbar.setMaximum(self.pbar.maximum() + 1)
-      self.progress_advance()
-    except Exception:
-      self.cancel_upload()
-      raise
+    offset = self.functions.pop()
+    func = instances.FunctionInstance(self.file_version_id, offset)
+    self.instance_set.append(func.serialize())
+
+    if len(self.instance_set) >= 100 or not self.functions:
+      q = network.QueryWorker("POST", "collab/instances/",
+                              params=self.instance_set, json=True)
+      q.start(self.progress_advance)
+      self.instance_set = []
+      self.pbar.setMaximum(self.pbar.maximum() + 1)
+    self.progress_advance()
 
   def progress_advance(self, result=None):
     del result
     new_value = self.pbar.value() + 1
-    self.pbar.setValue(new_value)
     if new_value >= self.pbar.maximum():
       self.pbar.accept()
-
-  def cancel_upload(self):
-    self.timer.stop()
-    self.timer = None
-    self.pbar = None
-
-  def reject_upload(self):
-    self.cancel_upload()
+    else:
+      self.pbar.setValue(new_value)
 
   def accept_upload(self):
-    self.cancel_upload()
+    self.clean()
+    self.delayed_queries = []
+
     self.start_task()
 
   def start_task(self):
```

Comment:
I'm not sure I follow. Is there something to fix here? Is there something to fix for master? Would you like me to rebase again and you'll comment then?